engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/integer/AddNode.java

```diff
@@ -55,6 +62,25 @@ Object doBigIntegerLong(EnsoBigInteger self, long that) {
     return self.asDouble() + that;
   }
 
+  @Specialization(guards = "isForeignNumber(iop, that)")
+  Object doInterop(
+      Object self,
+      TruffleObject that,
+      @CachedLibrary(limit = "3") InteropLibrary iop,
+      @Cached AddNode delegate) {
+    try {
+      if (iop.fitsInLong(that)) {
+        return delegate.execute(self, iop.asLong(that));
+      } else if (iop.fitsInDouble(that)) {
+        return delegate.execute(self, iop.asDouble(that));
+      } else if (iop.fitsInBigInteger(that)) {
+        return delegate.execute(self, new EnsoBigInteger(iop.asBigInteger(that)));
+      }
+    } catch (UnsupportedMessageException ex) {
+    }
```

Comment:
Shouldn't ignore the exception

Reply:
It isn't ignored (that much). It then calls into `doOther` which yields an exception.